ksql-cli/src/main/java/io/confluent/ksql/Ksql.java

```diff
@@ -73,25 +76,66 @@ public static void main(final String[] args) throws IOException {
     }
   }
 
-  @SuppressFBWarnings("OBL_UNSATISFIED_OBLIGATION")
-  @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
-  private static Properties loadProperties(final Optional<String> propertiesFile) {
-    final Properties properties = new Properties();
-    if (!propertiesFile.isPresent()) {
-      return properties;
-    }
+  void run() {
+    final Map<String, String> configProps = options.getConfigFile()
+        .map(Ksql::loadProperties)
+        .orElseGet(Collections::emptyMap);
 
-    try (InputStream input = new FileInputStream(propertiesFile.get())) {
-      properties.load(input);
-      if (properties.containsKey(KsqlConfig.KSQL_SERVICE_ID_CONFIG)) {
-        properties.put(
-            StreamsConfig.APPLICATION_ID_CONFIG,
-            properties.getProperty(KsqlConfig.KSQL_SERVICE_ID_CONFIG)
-        );
+    final Map<String, String> localProps = stripClientSideProperties(configProps);
+    final Map<String, String> clientProps = PropertiesUtil.applyOverrides(configProps, systemProps);
```

Comment:
might be worth stripping the systemProps of ssl configurations completely if the server is not using a secure protocol

Reply:
Do you mean remove ssl props from clientProps if the protocol is not https?  No point. The props won't be read anyway and they're not stored anywhere,